app/models/miq_worker/container_common.rb

```diff
@@ -29,6 +29,29 @@ def scale_deployment
       delete_container_objects if self.class.workers.zero?
     end
 
+    def patch_deployment
+      # Start with just resource constraints. Perhaps the livenessProbe, readinessProbe,
+      # and various timeouts such as terminationGracePeriodSeconds, could be patched later on.
+      # Note, we need to specify the name and image as they're required fields for the API to 'find'
+      # the correct container, even if we only ever have one.
+      data =
+        { :spec =>
+          { :template =>
+            { :spec =>
+              { :containers =>
+                [
+                  { :name      => worker_deployment_name,
+                    :image     => container_image,
+                    :resources => resource_constraints
+                  }
+                ]
+              }
+            }
+          }
+        }
```

Comment:
something about this alignment is weird to me...I think it should be 

```suggestion
      data = {
        :spec => {
          :template => {
            :spec => {
              :containers => [
                {
                  :name      => worker_deployment_name,
                  :image     => container_image,
                  :resources => resource_constraints
                }
              ]
            }
          }
        }
      }
```